src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -41,6 +41,7 @@
 
 # vprintln(args...) = println(args...)  # Uncomment for output
 vprintln(args...) = nothing
+check_nan(X, msg) = any(isnan, X) && error(msg)
```

Comment:
I would completely inline this function. It is basically no shorter if you make it explicit (which is clearer).